connection.go

```diff
@@ -351,6 +364,17 @@ func Connect(addr string, opts Opts) (conn *Connection, err error) {
 		}
 	}
 
+	conn.clientProtocolVersion = ClientProtocolVersion
+	conn.clientFeatures = ClientFeatures
+
+	if err = checkProtocolVersion(opts.RequiredProtocolVersion, conn.clientProtocolVersion); err != nil {
+		return nil, fmt.Errorf("client: %w", err)
+	}
+
+	if err = checkFeatures(opts.RequiredFeatures, conn.clientFeatures); err != nil {
+		return nil, fmt.Errorf("client: %w", err)
+	}
+
```

Comment:
It is also better to use a copy of the slice opts.RequiredFeatures.

Reply:
Well, to be honest, it seems more like an excessive copy since this is an internal method and we know what it may do, but let it be. Changed

Reply:
```Go
someFeaturesList = []Feature{Feature1, Feature2}
opts.RequiredFeatures = someFeatureList
conn := tarantool.Connect(conn, opts)
opts.RequiredFeatures[0] = Feature666 // It will change 
```

Reply:
Added test